pandaserver/dataservice/adder_atlas_plugin.py

```diff
@@ -237,324 +235,321 @@ def update_outputs(self):
+            try:
+                # check nevents
+                if file.type == "output" and not is_zip_file and not self.add_to_top_only and self.job.prodSourceLabel == "managed":
+                    if file.lfn not in self.extra_info["nevents"]:
+                        to_skip = False
+                        # exclude some formats
+                        for patt in ["TXT", "NTUP", "HIST"]:
+                            if file.lfn.startswith(patt):
+                                to_skip = True
+                        if not to_skip:
+                            err_msg = f"nevents is missing in jobReport for {file.lfn}"
+                            self.logger.warning(err_msg)
+                            self.job.ddmErrorCode = ErrorCode.EC_MissingNumEvents
+                            raise ValueError(err_msg)
+                    if file.lfn not in self.extra_info["guid"] or file.GUID != self.extra_info["guid"][file.lfn]:
+                        self.logger.debug(f"extra_info = {str(self.extra_info)}")
+                        err_msg = f"GUID is inconsistent between jobReport and pilot report for {file.lfn}"
+                        self.logger.warning(err_msg)
+                        self.job.ddmErrorCode = ErrorCode.EC_InconsistentGUID
+                        raise ValueError(err_msg)
+
+                # fsize
+                fsize = None
+                if file.fsize not in ["NULL", "", 0]:
+                    try:
+                        fsize = int(file.fsize)
+                    except Exception:
+                        error_type, error_value, _ = sys.exc_info()
+                        self.logger.error(f"{self.job_id} : {error_type} {error_value}")
+                # use top-level dataset name for alternative stage-out
+                if file.lfn not in self.job.altStgOutFileList():
+                    file_destination_dispatch_block = file.destinationDBlock
                 else:
-                    sub_to_ds_map[file.destinationDBlock] = file.dataset
-                # append to fileList
-                # commented since unused elsewhere
-                # fileList.append(file.lfn)
-                # add only log file for failed jobs
-                if self.job_status == "failed" and file.type != "log":
-                    continue
-                # add only log file for successful ES jobs
+                    file_destination_dispatch_block = file.dataset
+                # append to map
+                if file_destination_dispatch_block not in id_map:
+                    id_map[file_destination_dispatch_block] = []
+                file_attrs = {
+                    "guid": file.GUID,
+                    "lfn": file.lfn,
+                    "size": fsize,
+                    "checksum": file.checksum,
+                    "ds": file_destination_dispatch_block,
+                }
+                # add SURLs if LFC registration is required
+                if self.use_central_lfc():
```

Comment:
I find the use_central_lfc function confusing (i.e. LFC was something very old, not sure if Rucio is considered a central LFC and not sure this is related at all to LFCs at the moment). The function just checks the add_to_top_only variable - maybe we can do if directly against add_to_top_only and remove this function?